### PR TITLE
[FIX] Variables: Include variables defined in sub-directories

### PR DESCRIPTION
**Pull request by matz3**

### CODE REVIEW

File: lib/plugin/variable-collector.js [hunk 1]
```diff
@@ -43,15 +47,46 @@ VariableCollector.prototype = {
...
+				if (currentDirname === "." || currentDirname === "/") {
+					// Check for root (relative / absolute)
+					return false;
+				}
+				const libraryBaseFile = posixPath.join(currentDirname, baseFileName);
+				if (libraryBaseFile === rootFilename || aPosixImports.includes(libraryBaseFile)) {
+					return true;
+				} else {
+					// Check whether libraryBaseFile of parent directory is imported
+					return check(posixPath.join(currentDirname, ".."));
```
Comment (RandomByte) [inline lib/plugin/variable-collector.js:76]: path.dirname

Comment (RandomByte) [reply lib/plugin/variable-collector.js:76]: And maybe exit if `posixPath.dirname(currentDirname) === currentDirname`?

Comment (matz3) [reply lib/plugin/variable-collector.js]: Done

**Comment by matz3** [comment]
Decided to convert all paths to posix to be able to compare them independently of the OS and the paths used.

**Review by RandomByte — APPROVED** [review]

**Pull request merged**